chemprop/data/utils.py

```diff
@@ -298,19 +313,44 @@ def get_data(path: str,
         features_data = np.concatenate(features_data, axis=1)
     else:
         features_data = None
-        
+
     if phase_features_path is not None:
         phase_features = load_features(phase_features_path)
         for d_phase in phase_features:
             if not (d_phase.sum() == 1 and np.count_nonzero(d_phase) == 1):
                 raise ValueError('Phase features must be one-hot encoded.')
         if features_data is not None:
-            features_data = np.concatenate((features_data,phase_features), axis=1)
-        else: # if there are no other molecular features, phase features become the only molecular features
+            features_data = np.concatenate((features_data, phase_features), axis=1)
+        else:  # if there are no other molecular features, phase features become the only molecular features
             features_data = np.array(phase_features)
     else:
         phase_features = None
 
+    # Load constraints
+    if constraints_path is not None:
+        constraints_data = []
+        reader = pd.read_csv(constraints_path)
+        reader_columns = reader.columns.tolist()
+        if len(reader_columns) != len(set(reader_columns)):
+            raise ValueError(f'There are duplicates in {constraints_path}.')
+        for target in args.target_columns:
+            if target in reader_columns:
+                constraints_data.append(reader[target].values)
+            else:
+                constraints_data.append([None] * len(reader))
+        constraints_data = np.transpose(constraints_data)  # each is num_data x num_targets
+
+        if args.save_smiles_splits:
+            raw_constraints_data = []
+            for target in reader_columns:
+                raw_constraints_data.append(reader[target].values)
+            raw_constraints_data = np.transpose(raw_constraints_data)  # each is num_data x num_columns
+        else:
+            raw_constraints_data = None
```

Comment:
You don't have to, but may be worth considering breaking this out into a `get_constraints` function, like the `get_data_weights` function or similar.

Reply:
Thanks for this suggestion. By doing so, the code is more clear.